Exp10.hpp msvc issue

I think the little adaptor isnt quite right on windows, the specializations dont exist

```
    TestExp.cpp
C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp10.hpp(24,18): error C2039: 'exp10f': is not a member of '`global namespace'' (compiling source file C:\build\PothosSDR_vc16\Source\PothosComms\math\TestExp.cpp) [C:\build\PothosSDR_vc16\Build\PothosComms\math\MathBlocks.vcxproj] [C:\build\PothosSDR_vc16\PothosComm
s.vcxproj]
C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp10.hpp(24,1): error C3861: 'exp10f': identifier not found (compiling source file C:\build\PothosSDR_vc16\Source\PothosComms\math\TestExp.cpp) [C:\build\PothosSDR_vc16\Build\PothosComms\math\MathBlocks.vcxproj] [C:\build\PothosSDR_vc16\PothosComms.vcxproj]
C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp10.hpp(24,18): error C2039: 'exp10f': is not a member of '`global namespace'' (compiling source file C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp.cpp) [C:\build\PothosSDR_vc16\Build\PothosComms\math\MathBlocks.vcxproj] [C:\build\PothosSDR_vc16\PothosComms.vc
xproj]
C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp10.hpp(30,18): error C2039: 'exp10': is not a member of '`global namespace'' (compiling source file C:\build\PothosSDR_vc16\Source\PothosComms\math\TestExp.cpp) [C:\build\PothosSDR_vc16\Build\PothosComms\math\MathBlocks.vcxproj] [C:\build\PothosSDR_vc16\PothosComms
.vcxproj]
C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp10.hpp(24,1): error C3861: 'exp10f': identifier not found (compiling source file C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp.cpp) [C:\build\PothosSDR_vc16\Build\PothosComms\math\MathBlocks.vcxproj] [C:\build\PothosSDR_vc16\PothosComms.vcxproj]
C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp10.hpp(30,18): error C2039: 'exp10': is not a member of '`global namespace'' (compiling source file C:\build\PothosSDR_vc16\Source\PothosComms\math\Exp.cpp) [C:\build\PothosSDR_vc16\Build\PothosComms\math\MathBlocks.vcxproj] [C:\build\PothosSDR_vc16\PothosComms.vcx
proj]
```

--- a/math/CMakeLists.txt
+++ b/math/CMakeLists.txt
@@ -14,12 +14,25 @@ if(MSVC)
     add_definitions(/bigobj) #arithmetic too large in debug mode
 endif(MSVC)
 
+# exp10 and exp10f are GNU extensions, so we know MSVC won't have them.
+if(NOT MSVC)
+    include(CheckSymbolExists)
+    set(CMAKE_REQUIRED_DEFINITIONS -D_GNU_SOURCE)
+    set(CMAKE_REQUIRED_LIBRARIES m)
+    check_symbol_exists(exp10f "math.h" HAVE_EXP10F)
+    check_symbol_exists(exp10 "math.h" HAVE_EXP10)
+endif()
+configure_file(
+    Exp10.hpp.in
+    ${CMAKE_CURRENT_BINARY_DIR}/Exp10.hpp)
+
+include_directories(
+    ${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR})
+
 set(libraries CommsFunctions)
 
 if(xsimd_FOUND)
-    include_directories(
-        ${CMAKE_CURRENT_SOURCE_DIR}
-        ${CMAKE_CURRENT_BINARY_DIR})
     add_subdirectory(SIMD)
     list(APPEND libraries CommsMathSIMD)
 endif()

--- a/math/Exp10.hpp.in
+++ b/math/Exp10.hpp.in
@@ -3,32 +3,34 @@
 
 #pragma once
 
+#cmakedefine HAVE_EXP10F
+#cmakedefine HAVE_EXP10
+
 #include <cmath>
 
 namespace detail
 {
-    //
-    // There is no exp10 STL function for some reason, so
-    // abstract it on this layer.
-    //
-
     template <typename T>
     inline T exp10(T in)
     {
         return T(std::pow(T(10), in));
     }
 
+#ifdef HAVE_EXP10F
     template <>
     inline float exp10(float in)
     {
         return ::exp10f(in);
     }
+#endif
 
+#ifdef HAVE_EXP10
     template <>
     inline double exp10(double in)
     {
         return ::exp10(in);
     }
+#endif
 }
 
 template <typename T>

